apps/dashboard/test/system/batch_connect_widgets_test.rb

```diff
@@ -431,4 +431,116 @@ def make_bc_app(dir, form)
       end
     end
   end
+
+  test 'data-options-for-' do
+    Dir.mktmpdir do |dir|
+      "#{dir}/app".tap { |d| Dir.mkdir(d) }
+      SysRouter.stubs(:base_path).returns(Pathname.new(dir))
+      stub_scontrol
+      stub_sacctmgr
+      stub_git("#{dir}/app")
+
+      form = <<~HEREDOC
+        ---
+        form:
+          - cluster
+          - node_type
+        attributes:
+          cluster:
+            widget: "select"
+            options:
+              - owens
+              - pitzer
+          node_type:
+            widget: "select"
+            options:
+              - standard
+              - ['gpu', 'gpu', data-option-for-cluster-pitzer: false]
+      HEREDOC
+
+      Pathname.new("#{dir}/app/").join('form.yml').write(form)
+      base_id = 'batch_connect_session_context_path'
+
+      visit new_batch_connect_session_context_url('sys/app')
+
+      # owens is selected, standard and gpu are both visible
+      select('owens', from: 'batch_connect_session_context_cluster')
+      options = find_all("#batch_connect_session_context_node_type option")
+
+      assert_equal "standard", options[0]["innerHTML"]
+      assert_equal "gpu", options[1]["innerHTML"]
+
+      # select gpu, to test that it's deselected properly when pitzer is selected
+      select('gpu', from: 'batch_connect_session_context_node_type')
+
+      # pitzer is selected, gpu is not visible
+      select('pitzer', from: 'batch_connect_session_context_cluster')
+      options = find_all("#batch_connect_session_context_node_type option")
+      
+      display_property = { "display" => "none" }
+
+      assert_equal "standard", options[0]["innerHTML"]
+      assert_equal display_property, options[1].style('display')
```

Comment:
The `visible?` API works now. I know it hasn't in the past, but at some point it started to work reliably. 

https://github.com/OSC/ondemand/blob/a9569326c1e51967f47b670f3163b307a5fab437/apps/dashboard/test/system/batch_connect_widgets_test.rb#L70

Reply:
`visible?` is returning `true` here even when `display = none` is true.

Reply:
🤦‍♂️ OK, maybe it's not reliable.  

In that case, can we do something like this so it's consistent with other tests?

https://github.com/OSC/ondemand/blob/ccfb597ed7a5a2d961c388195333304bf22dc501/apps/dashboard/test/system/batch_connect_test.rb#L610